File: src/typescript/logic/wrong_location/writeAndRead.ts
import { writeFile, readFile, unlink } from 'fs';


function readFromDatabase(callback: (data: string) => void) {
    setTimeout(() => {
        const simulatedData = 'Hello from the Database!';
        callback(simulatedData);
    }, 1000);
}

function writeAndRead() {
    readFromDatabase((data) => {
        writeFile('message.txt', data, (err) => {
            if (err) {
                console.error('Write Error:', err);
            } else {
                console.log('File written successfully.');
            }
        });
    })

    readFile('message.txt', 'utf8', (err, content) => {
        if (err) {
            console.error('Read Error:', err);
        } else {
            console.log('File Content:', content);
        }
    });
}

writeAndRead();
\ No newline at end of file

File: truth_set/python/logic/wrong_location/writeAndRead.py
import asyncio
import aiofiles

async def read_from_database():
    await asyncio.sleep(3)  # Simulating a large database call
    return "Hello from the Database!"

async def write_file(data):
    async with aiofiles.open('message.txt', 'w') as f:
        await f.write(data)
        print("Write in progress...")
        await asyncio.sleep(5)  # Simulate a slow write process
        print("Write completed.")

async def read_file():
    try:
        async with aiofiles.open('message.txt', 'r') as f:
            print("Attempting to read file...")
            content = await f.read()
            print(f"File Content: {content}")
    except FileNotFoundError:
        print("File not found during read.")

async def write_and_read():
    data = await read_from_database()
    
    write_task = asyncio.create_task(write_file(data))
    read_task = asyncio.create_task(read_file())
    
    await asyncio.gather(write_task, read_task)

async def main():
    await write_and_read()

if __name__ == "__main__":
    asyncio.run(main())
\ No newline at end of file